### ISSUE TITLE
Releasing statement in a finalizer (synchronous transaction)

### ISSUE THREAD

**Issue by quickquestion** [Raizlabs/DBFlow#1156]
DBFlow 4.0.0-beta5:
Issue Kind: Question

Description: 
I want to save more than one model/table in a single synchronous transaction (in a service, i am using sqlCipher wrapper):
```
FlowManager.getDatabase(AppDatabase.class).executeTransaction(new ITransaction() {
                        @Override
                        public void execute(DatabaseWrapper databaseWrapper) {
                            Delete.tables(Product.class, ProductCategory.class, Company.class);
                            for (Company c : x.getCompanies())
                            {
                                c.save(databaseWrapper);
                            }
                            for (Product product : x.getProducts())
                            {
                                product.save(databaseWrapper);
                            }
                        }
                    });
```
Code is working, i.e. i can read records after insertion, but i am having same Logcat erros as described in [https://github.com/Raizlabs/DBFlow/issues/1050](url)
i.e.:
`Releasing statement in a finalizer. Please ensure that you explicitly call close() on your cursor: INSERT INTO `Product`(`Id`,`ProductCategoryId`,`CreatedDate`,`CreatedBy`,`UpdatedDate`,`UpdatedBy`,`
                                                                      net.sqlcipher.database.DatabaseObjectNotClosedException: Application did not close the cursor or database object that was opened here
                                                                          at net.sqlcipher.database.SQLiteCompiledSql.<init>(SQLiteCompiledSql.java:62)
                                                                          at net.sqlcipher.database.SQLiteProgram.<init>(SQLiteProgram.java:103)
                                                                          at net.sqlcipher.database.SQLiteStatement.<init>(SQLiteStatement.java:39)
                                                                          at net.sqlcipher.database.SQLiteDatabase.compileStatement(SQLiteDatabase.java:1556)
                                                                          at com.raizlabs.dbflow.android.sqlcipher.SQLCipherDatabase.compileStatement(SQLCipherDatabase.java:55)
                                                                          at com.raizlabs.android.dbflow.structure.ModelAdapter.getInsertStatement(ModelAdapter.java:73)
                                                                          at com.raizlabs.android.dbflow.sql.saveable.ModelSaver.save(ModelSaver.java:35)
                                                                          at com.raizlabs.android.dbflow.structure.ModelAdapter.save(ModelAdapter.java:124)
                                                                          at com.raizlabs.android.dbflow.structure.BaseModel.save(BaseModel.java:67)
                                                                          at XXX.RestClient$1$1.execute(IfmRestClient.java:99)
                                                                          at com.raizlabs.android.dbflow.config.DatabaseDefinition.executeTransaction(DatabaseDefinition.java:218) .......`

I saw your recommendation about ProcessModelTransaction but i am not sure how to use it to store multiple tables in a single synchronous transaction?

Thank you in advance for your help

**Comment by joshln87** [comment]
is there any updates to this? facing this as well

**Comment by baranowskinet** [comment]
I also have the same problem especially when large amount of records I want to update. Any recommend pattern for update many tables in one transaction?

**Comment by xinghui322** [comment]
Call save() not save(databaseWrapper), save(databaseWrapper) didn't close the newly compiled statement. This maybe a bug @agrosner

**Comment by baranowskinet** [comment]
but what with transaction if I with just call save()?

**Comment by agrosner** [comment]
this is fixed for 4.0.5
